mesonbuild/coredata.py

```diff
@@ -576,6 +578,7 @@ def __init__(self, options: argparse.Namespace, scratch_dir: str, meson_command:
             DependencyCache(self.options, MachineChoice.HOST))
 
         self.compiler_check_cache: T.Dict['CompilerCheckCacheKey', 'CompileResult'] = OrderedDict()
+        self.find_library_cache: T.Dict['FindLibraryCacheKey', 'FindLibraryResult'] = OrderedDict()
```

Comment:
This is going to mix together libraries from host and build machine right? I think we need to use a `PerMachine` here so that we can avoid that. I'd prefer to the `PerMachine` to adding the machine to the cache key.

Reply:
My patch just moves the existing in-process cache to persistence between runs. I think your concern would also apply to the existing cache?